src/main/java/com/exadel/sandbox/booking/service/BookingServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.exadel.sandbox.booking.service;
+
+import com.exadel.sandbox.address.dto.AddressResponseDto;
+import com.exadel.sandbox.address.entity.Address;
+import com.exadel.sandbox.booking.dto.BookingCreateDto;
+import com.exadel.sandbox.booking.dto.BookingResponseDto;
+import com.exadel.sandbox.booking.dto.BookingUpdateDto;
+import com.exadel.sandbox.booking.entity.Booking;
+import com.exadel.sandbox.booking.repository.BookingRepository;
+import com.exadel.sandbox.employee.dto.employeeDto.EmployeeResponseDto;
+import com.exadel.sandbox.employee.entity.Employee;
+import com.exadel.sandbox.exception.EntityNotFoundException;
+import com.exadel.sandbox.parking_spot.dto.ParkingSpotResponseDto;
+import com.exadel.sandbox.parking_spot.entity.ParkingSpot;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class BookingServiceImpl implements BookingService {
+
+    private final BookingRepository bookingRepository;
+    private final ModelMapper mapper;
+
+    @Override
+    public List<BookingResponseDto> getBookings() {
+        List<Booking> bookings = bookingRepository.findAll();
+        List<BookingResponseDto> bookingResponseDtos = new ArrayList<>();
+        for (Booking booking : bookings) {
+            bookingResponseDtos.add(fullMap(booking));
+        }
+        return bookingResponseDtos;
+    }
+
+    @Override
+    public BookingResponseDto getBookingById(Long id) {
+        Optional<Booking> bookingByID = bookingRepository.findById(id);
+        Booking booking = bookingByID.orElseThrow(
+                () -> new EntityNotFoundException("Can\'t get Booking with ID: " + id + ". Doesn\'t exist."));
+        return fullMap(booking);
+    }
+
+    @Override
+    public BookingResponseDto create(BookingCreateDto bookingCreateDto) {
+        Booking booking = mapper.map(bookingCreateDto, Booking.class);
+
+        if(bookingCreateDto.getParkingSpotResponseDto() != null)
+            booking.setParkingSpot(mapper.map(bookingCreateDto.getParkingSpotResponseDto(), ParkingSpot.class));
+
+        if(bookingCreateDto.getEmployeeResponseDto() != null)
+            booking.setEmployee(mapper.map(bookingCreateDto.getEmployeeResponseDto(), Employee.class));
+
+        return fullMap(bookingRepository.save(booking));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        Booking booking = bookingRepository.findById(id).orElseThrow(
```

Comment:
I think you don't need to create booking object. 
   bookingRepository.findById(id).orElseThrow(
                () -> new EntityNotFoundException("Can\'t delete Booking with ID: " + id + ". Doesn\'t exist."));
        bookingRepository.deleteById(id);

Reply:
Should I do the same for Address, Parking and ParkingSpot services too?

Reply:
I changed the code so that the object isn't created but it still checks if object exists with provided ID.